src/types/UiaTextRangeBase.cpp

```diff
@@ -102,14 +102,18 @@ void UiaTextRangeBase::_outputObjectState()
 #endif // _DEBUG
 
 // degenerate range constructor.
-UiaTextRangeBase::UiaTextRangeBase(_In_ IUiaData* pData, _In_ IRawElementProviderSimple* const pProvider) :
-    _cRefs{ 1 },
-    _pProvider{ THROW_HR_IF_NULL(E_INVALIDARG, pProvider) },
-    _start{ 0 },
-    _end{ 0 },
-    _degenerate{ true },
-    _pData{ THROW_HR_IF_NULL(E_INVALIDARG, pData) }
+#pragma warning(suppress : 26434) // WRL RuntimeClassInitialize base is a no-op and we need this for MakeAndInitialize
+HRESULT UiaTextRangeBase::RuntimeClassInitialize(_In_ IUiaData* pData, _In_ IRawElementProviderSimple* const pProvider) noexcept
 {
+    RETURN_HR_IF_NULL(E_INVALIDARG, pProvider);
+    RETURN_HR_IF_NULL(E_INVALIDARG, pData);
+
+    _pProvider = pProvider;
+    _start = 0;
+    _end = 0;
+    _degenerate = true;
+    _pData = pData;
+
     _id = id;
     ++id;

```

Comment:
I wonder if this is going to come back to bite us at some point. A **program-global** static ever-incrementing ID number?

Reply:
It doesn't seem like we use _id anywhere, really, other than for debugging purposes. So we should be good.